app/frontend/admin/courses/CourseEditor__Root.res

```diff
@@ -162,8 +171,77 @@ let reducer = (state, action) =>
   | UpdateCourse(course) =>
     let newCourses = Pagination.update(state.courses, Course.updateList(course))
     {...state, courses: newCourses}
+  | ReorderCourses(newCourses) => {...state, courses: Pagination.FullyLoaded(newCourses)}
   }
 
+let loadCourses = (courseId, state, cursor, ~skipSchoolStatsLoad=true, send) => {
+  let variables = CoursesQuery.makeVariables(
+    ~status=?state.filter.status,
+    ~after=?cursor,
+    ~search=?state.filter.name,
+    ~courseId=Belt.Option.getWithDefault(courseId, ""),
+    ~skipCourseLoad=Belt.Option.isNone(courseId),
+    ~skipSchoolStatsLoad,
+    (),
+  )
+  CoursesQuery.fetch(variables)
+  |> Js.Promise.then_((response: CoursesQuery.t) => {
+    let responseCourses = Belt.SortArray.stableSortBy(response.courses.nodes, (a, b) =>
+      a.sortIndex - b.sortIndex
+    )
+    let courses = responseCourses->Js.Array2.map(rawCourse => Course.makeFromFragment(rawCourse))
+    let course = response.course->Belt.Option.map(Course.makeFromFragment)
+    switch course {
+    | None => send(UpdateSelectedCourse(None))
+    | Some(course) =>
+      switch courses->Js.Array2.find(c => {c.id == course.id}) {
+      | None => send(UpdateSelectedCourse(Some(course)))
+      | Some(_) => send(UpdateSelectedCourse(None))
+      }
+    }
+
+    let schoolStats = Belt.Option.map(response.schoolStats, s => Loaded({
+      studentsCount: s.studentsCount,
+      coachesCount: s.coachesCount,
+    }))
+    send(
+      LoadCourses(
+        response.courses.pageInfo.endCursor,
+        response.courses.pageInfo.hasNextPage,
+        courses,
+        response.courses.totalCount,
+        schoolStats,
+      ),
+    )
+    Js.Promise.resolve()
+  })
+  |> ignore
+}
+
+let handleMoveCourse = (~course, ~direction: Course.direction, ~send, ~state) => {
+  let id = Course.id(course)
+  MoveCourseQuery.fetch(
+    ~notify=false,
+    {
+      id,
+      direction: switch direction {
+      | Up => #Up
+      | Down => #Down
+      },
+    },
+  )
+  |> Js.Promise.then_(_ => {
```

Comment:
Use `Pipe(->)`  with `Js.promise2`  instead of the deprecated `Triangle Pipe (|>)`  with `Js.promise`  

https://rescript-lang.org/docs/manual/v10.0.0/promise#jspromise2